source/concepts/bucket-versioning.rst

```diff
@@ -10,78 +10,309 @@ Bucket Versioning
+return the ``DeleteMarker`` back as part of the response. Similarly, performing
+a ``LIST`` operation by default returns only objects which are *not* a
+``DeleteMarker``.
 
-<Diagram>
+To permanently delete an object version, perform the ``DELETE`` operation and 
+specify the version ID of the object to delete. Versioned delete operations 
+are **irreversible**.
 
-MinIO generates a unique immutable ID for each object. If a ``PUT`` request
-contains an object name which duplicates an existing object, MinIO does *not*
-overwrite the "older" object. Instead, MinIO retains all object versions while
-considering the most recently written "version" of the object as "latest".
-Applications retrieve the latest object version by default, but *may* retrieve
-any other version in the history of that object. To view all versions of an
-object or objects in a bucket, use the :mc-cmd-option:`mc ls versions` command.
+.. card-carousel:: 1
 
-By default, deleting an object does *not* remove all existing versions of
-that object. Instead, MinIO places a "delete" marker for the object, such that
-applications requesting only the latest object versions do not see the object.
-Applications *may* retrieve earlier versions of that object. To completely
-delete an object and its entire version history, use the
-:mc-cmd-option:`mc rm versions` command. 
+   .. card:: Deleting an Object
+
+      .. image:: /images/retention/minio-versioning-delete-object.svg
+         :alt: Deleting an Object
+         :align: center
+
+      Performing a ``DELETE`` operation on a versioned object produces a 
+      ``DeleteMarker`` for that object.
+
+   .. card:: Reading a Deleted Object
+
+      .. image:: /images/retention/minio-versioning-retrieve-deleted-object.svg
+         :alt: Object with Multiple Versions
+         :align: center
+
+      Clients by default retrieve the "latest" object version. MinIO returns
+      a ``404``-like response if the latest version is a ``DeleteMarker``.
+
+   .. card:: Retrieve Previous Version of Deleted Object
+
+      .. image:: /images/retention/minio-versioning-retrieve-version-before-delete.svg
+         :alt: Retrieve Version of Deleted Object
+         :align: center
+
+      Clients can retrieve any previous version of the object by specifying the
+      version ID, even if the "Latest" version is a ``DeleteMarker``.
+
+   .. card:: Delete a Specific Object Version
+
+      .. image:: /images/retention/minio-versioning-delete-specific-version.svg
+         :alt: Retrieve Version of Deleted Object
+         :align: center
+
+      Clients can delete a specific object version by specifying the version ID
+      as part of the ``DELETE`` operation. Deleting a specific version is 
+      **permanent** and does not result in the creation of a ``DeleteMarker``.
+
+The following :mc:`mc` commands operate on ``DeleteMarkers`` or versioned 
+objects:
+
+- Use :mc-cmd-option:`mc ls versions` to view all versions of an object,
```

Comment:
So under the hood there is a horrible blob of python code where, if you use `:mc-cmd-option:`, I shoehorn a `--` onto the last chunk of the argument. So `--` gets appended on automagically. But it's horrible, ugly, kludgy magic. 

This is wholly unintuitive (as it turns out), but refactoring and fixing this would take me a little while, so I've been putting it off :)